src/support.js

```diff
@@ -189,28 +193,354 @@ const supportTable = [
+ *
+ * @typedef {number | boolean | string} OptionValue
+ * @typedef {OptionValue | Array<{ since: string, value: OptionValue}>} OptionValueInfo
+ *
+ * @typedef {Object} OptionRedirectInfo
+ * @property {string} option
+ * @property {OptionValue} value
+ *
+ * @typedef {Object} OptionRangeInfo
+ * @property {number} start - recommended range start
+ * @property {number} end - recommended range end
+ * @property {number} step - recommended range step
+ *
+ * @typedef {Object} OptionChoiceInfo
+ * @property {boolean | string} value - boolean for the option that is originally boolean type
+ * @property {string?} description - undefined if redirect
+ * @property {string?} since - undefined if available since the first version of the option
+ * @property {string?} deprecated - deprecated since version
+ * @property {OptionValueInfo?} redirect - redirect deprecated value
+ */
+/** @type {OptionInfo[]} */
+const supportOptions = [
+  {
+    name: "arrowParens",
+    since: "1.9.0",
+    category: CATEGORY_JAVASCRIPT,
+    type: "choice",
+    default: "avoid",
+    description: "Include parentheses around a sole arrow function parameter.",
+    choices: [
+      {
+        value: "avoid",
+        description: "Omit parens when possible. Example: `x => x`"
+      },
+      {
+        value: "always",
+        description: "Always include parens. Example: `(x) => x`"
+      }
+    ]
+  },
+  {
+    name: "bracketSpacing",
+    since: "0.0.0",
+    category: CATEGORY_JAVASCRIPT,
+    type: "boolean",
+    default: true,
+    description: "Print spaces between brackets.",
+    oppositeDescription: "Do not print spaces between brackets."
+  },
+  {
+    name: "cursorOffset",
+    since: "1.4.0",
+    category: CATEGORY_SPECIAL,
+    type: "int",
+    default: -1,
+    range: { start: -1, end: Infinity, step: 1 },
+    description: dedent`
+      Print (to stderr) where a cursor at the given position would move to after formatting.
+      This option cannot be used with --range-start and --range-end.
+    `
+  },
+  {
+    name: "filepath",
+    since: "1.4.0",
+    category: CATEGORY_SPECIAL,
+    type: "path",
+    default: undefined,
+    description:
+      "Specify the input filepath. This will be used to do parser inference."
+  },
+  {
+    name: "insertPragma",
+    since: "1.8.0",
+    category: CATEGORY_SPECIAL,
+    type: "boolean",
+    default: false,
+    description: "Insert @format pragma into file's first docblock comment."
+  },
+  {
+    name: "jsxBracketSameLine",
+    since: "0.17.0",
+    category: CATEGORY_JAVASCRIPT,
+    type: "boolean",
+    default: false,
+    description: "Put > on the last line instead of at a new line."
+  },
+  {
+    name: "parser",
+    since: "0.0.10",
+    category: CATEGORY_GLOBAL,
+    type: "choice",
+    default: "babylon",
+    description: "Which parser to use.",
+    choices: [
+      { value: "flow", description: "Flow" },
+      { value: "babylon", description: "JavaScript" },
```

Comment:
Can you reorder flow and babylon? Just to keep the same order from the docs and the playground?

(not that it makes that much difference but still...)